polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1256,31 +1330,359 @@ impl<T: Config> Pallet<T> {
 		let value = (origin_location, assets.into_inner());
 		ensure!(T::XcmTeleportFilter::contains(&value), Error::<T>::Filtered);
 		let (origin_location, assets) = value;
+		for asset in assets.iter() {
+			let transfer_type =
+				<T::XcmExecutor as AssetTransferSupport>::determine_for(asset, &dest)
```

Comment:
```suggestion
				T::XcmExecutor::determine_for(asset, &dest)
```

it works without `<T::XcmExecutor as AssetTransferSupport>`, I dont know if we want to be explicit here